spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -249,63 +273,19 @@ protected MessageProducer createConsumerEndpoint(String name, String group, Queu
 		return adapter;
 	}
 
-	@Override
-	protected void afterUnbindConsumer(String name, String group,
-			ExtendedConsumerProperties<RabbitConsumerProperties> consumerProperties) {
-		cleanAutoDeclareContext(consumerProperties.getExtension().getPrefix(), baseQueueName(name, group));
-	}
-
-	@Override
-	protected Queue createConsumerDestinationIfNecessary(String name, String group,
-			ExtendedConsumerProperties<RabbitConsumerProperties> properties) {
-		boolean anonymous = !StringUtils.hasText(group);
-		String baseQueueName = anonymous ? groupedName(name, ANONYMOUS_GROUP_NAME_GENERATOR.generateName())
-				: groupedName(name, group);
-		if (this.logger.isInfoEnabled()) {
-			this.logger.info("declaring queue for inbound: " + baseQueueName + ", bound to: " + name);
-		}
-		String prefix = properties.getExtension().getPrefix();
-		String exchangeName = applyPrefix(prefix, name);
-		Exchange exchange = buildExchange(properties.getExtension(), exchangeName);
-		if (properties.getExtension().isDeclareExchange()) {
-			declareExchange(exchangeName, exchange);
-		}
-		String queueName = applyPrefix(prefix, baseQueueName);
-		boolean partitioned = !anonymous && properties.isPartitioned();
-		boolean durable = !anonymous && properties.getExtension().isDurableSubscription();
-		Queue queue;
-
-		if (anonymous) {
-			queue = new Queue(queueName, false, true, true, queueArgs(queueName, properties.getExtension(), false));
+	private String deadLetterExchangeName(RabbitCommonProperties properties) {
+		if (properties.getDeadLetterExchange() == null) {
+			return properties.getPrefix() + RabbitCommonProperties.DEAD_LETTER_EXCHANGE;
 		}
 		else {
-			if (partitioned) {
-				String partitionSuffix = "-" + properties.getInstanceIndex();
-				queueName += partitionSuffix;
-			}
-			if (durable) {
-				queue = new Queue(queueName, true, false, false,
-						queueArgs(queueName, properties.getExtension(), false));
-			}
-			else {
-				queue = new Queue(queueName, false, false, true,
-						queueArgs(queueName, properties.getExtension(), false));
-			}
-		}
-		declareQueue(queueName, queue);
-		if (properties.getExtension().isBindQueue()) {
-			declareConsumerBindings(name, properties, exchange, partitioned, queue);
-		}
-		if (durable) {
-			autoBindDLQ(applyPrefix(properties.getExtension().getPrefix(), baseQueueName), queueName,
-					properties.getExtension());
+			return properties.getDeadLetterExchange();
 		}
-		return queue;
 	}
 
-	private String baseQueueName(String name, String group) {
-		return !StringUtils.hasText(group) ? groupedName(name, ANONYMOUS_GROUP_NAME_GENERATOR.generateName())
-				: groupedName(name, group);
+	@Override
+	protected void afterUnbindConsumer(ConsumerDestination consumerDestination, String group,
+			ExtendedConsumerProperties<RabbitConsumerProperties> consumerProperties) {
+		((RabbitExchangeQueueProvisioner)provisioningProvider).cleanAutoDeclareContext(consumerDestination.getName());
```

Comment:
Doh! I see now:
```
super(true, new String[0], provisioningProvider);
```
So, we still can come back to the generic arg and don't have this ugly cast 😄